tests/projects/test_projects.py

```diff
@@ -161,6 +161,46 @@ def test_run_local_git_repo(tmpdir, local_git_repo,
         assert metric.value == expected_metrics[metric.key]
 
 
+@pytest.mark.parametrize("use_start_run", map(str, [0, 1]))
```

Comment:
Thanks for adding the test! Could we actually just modify the existing `test_run_local_git_repo` test to pass a `version` argument when calling `projects.run`? I think doing that plus adding a unit test for `_is_valid_branch_name` will give us good coverage. If we want we could actually parametrize `test_run_local_git_repo` via `@pytest.mark.parametrize("version", [None, "master"])` to test both cases (when a version is specified and when a version isn't specified). We'd then need an if/else based on `version` to conditionally either check that the mlflow.gitBranchName tag was/wasn't set

Reply:
The reason behind adding as another test is because there are two ways to run a local git project. One is `mlflow run /Users/xxx/mlflow/examples/sklearn_elasticnet_wine -P alpha=0.2` and the other is `mlflow run file:///Users/xxx/mlflow#examples/sklearn_elasticnet_wine -P alpha=0.2`. Note the difference in the uri, with or without the `file:` prefix and the sub directory proceeded by `#`. Without the `file:` prefix, the `version` is not allowed. So the `test_local_git_repo` is for the first uri case and the `test_local_git_repo_with_version` is for the second uri case with version option.

But anyway, I converted them into one with your suggestion.